# This file is part of HiGFXback

# requires
REQUIRES="make-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
sh ./Configure --disable-static --prefix

# build
make

# install
install -d $DESTDIR/bin
install rhash_shared $DESTDIR/bin/rhash
install -d $DESTDIR/include
install -m 644 librhash/rhash.h $DESTDIR/include
install -d $DESTDIR/lib
install librhash/librhash.so.0 $DESTDIR/lib
ln -sf librhash.so.0 $DESTDIR/lib/librhash.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 dist/librhash.pc $DESTDIR/lib/pkgconfig

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/rhash-build.pc << EOF
Name: RHash
Version: 1.3.5
Description: Recursive Hasher library and utility for checking file hash sums
Requires: $REQUIRES

devel=\\
/include/rhash.h \\
/lib/librhash.so \\
/lib/pkgconfig/librhash.pc

exec=\\
/bin/rhash \\
/lib/librhash.so.0
EOF

exit
--- RHash-1.3.5.orig/Configure
+++ RHash-1.3.5/Configure
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+PREFIX=/usr/local
+TARGET=\$\(TARGET\)
+LIBRARY=\$\(LIBRARY\)
+
+while : ; do
+  case "$1" in
+    "")
+      break
+    ;;
+    --prefix)
+      PREFIX=$2
+      shift
+    ;;
+    --disable-static)
+      TARGET=\$\(SHARED_TRG\)
+      LIBRARY=\$\(SONAME\)
+    ;;
+    *)
+      echo Unknown option \"$1\".
+      exit 1
+    ;;
+  esac
+  shift
+done
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" Makefile
+
+sed -i "/^all/s|:.*|: $TARGET|" Makefile
+
+sed -i "/^PREFIX/s|=.*|= $PREFIX|" librhash/Makefile
+
+sed -i "/^all/s|:.*|: $LIBRARY|" librhash/Makefile
+
+sed -i "/^prefix/s|=.*|=$PREFIX|" dist/librhash.pc
--- RHash-1.3.5.orig/dist/librhash.pc
+++ RHash-1.3.5/dist/librhash.pc
@@ -0,0 +1,11 @@
+prefix=/usr/local
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include
+
+Name: librash
+Description: LibRHash library
+Version: 1.3.5
+Cflags: -I${includedir}
+Libs: -L${libdir} -lrhash
+Libs.private: -ldl
